src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/NativeMemoryTests.cs

```diff
@@ -558,5 +558,43 @@ public void ClearWithNullPointerAndZeroByteCountTest()
 
             // This test method just needs to check that no exceptions are thrown
         }
+
+        [Fact]
+        public void CopyNullBlockShouldNoOp()
+        {
+            // This should not throw
+            NativeMemory.Copy(null, null, 0);
+        }
+
+        [Fact]
+        public void CopyEmptyBlockShouldNoOp()
+        {
+            int* source = stackalloc int[1] { 42 };
+            int* destination = stackalloc int[1] { 0 };
+
+            NativeMemory.Copy(source, destination, 0);
+
+            Assert.Equal(0, destination[0]);
+        }
+
+        [Theory]
+        [InlineData(1, 1, 1)]
+        [InlineData(1, 16, 1)]
+        [InlineData(16, 16, 16)]
+        [InlineData(1024, 16, 16)]
+        public void CopyTest(int sourceSize, int destinationSize, int byteCount)
+        {
+            void* source = NativeMemory.AllocZeroed((nuint)sourceSize);
+            void* destination = NativeMemory.AllocZeroed((nuint)destinationSize);
+
+            new Span<byte>(source, sourceSize).Fill(0b10101010);
+
+            NativeMemory.Copy(source, destination, (nuint)byteCount);
+
+            Equals(byteCount - 1, new Span<byte>(destination, destinationSize).LastIndexOf<byte>(0b10101010));
+
+            NativeMemory.Free(source);
+            NativeMemory.Free(destination);
+        }
     }
```

Comment:
what is expected behavior for overlaps? it is unclear to me from the discussion in the issue, but it seemed that it was undesirable that it be undefined or crash, in which case we should have tests presumably.

Reply:
According to discussion, `Copy` follows behavior of `memset` native implementation. In case of overlapping, the behavior is undefined. I can reflect this in the documentation.